pkg/common/proto/agent.proto

```diff
@@ -0,0 +1,48 @@
+syntax = "proto3";
+
+option go_package = ".;pb";
+
+service AgentService{
+    rpc PerformFantasyActions(FantasyLandscape) returns (FantasySelections) {}
+}
+
+message FantasyLandscape{
+    int32 matchNumber = 1;
+    LeagueSettings settings = 2;
+    FantasyTeam botTeam = 3;
+    repeated Player players = 4;
+    /* TODO historical / statistical data */
+    /* TODO competition lineups */
+}
+
+message FantasySelections{
+    repeated PlayerSlot slots = 1;
+}
+
+message FantasyTeam{
+    string id = 1;
+    string name = 2;
+    string owner = 3;
+}
+
+message LeagueSettings{
+    int32 numTeams = 1;
+    repeated PlayerSlot slotsPerTeam = 2;
+    /* scoring, budgets, etc. */
+}
+
+message PlayerSlot{
+    string name = 1;
+    bool canBeEmpty = 2;
+    repeated string allowedPlayerPositions = 4;
+    optional string assignedPlayerId = 3;
+}
+
+message Player{
+    string id = 1;
+    string fullName = 2;
+    repeated string allowedPositions = 3;
+    string team = 4; /* professional team, not fantasy team */
+    string status = 5;
+    optional int32 fantasyTeamId = 6; /* 0 should be unassigned */
```

Comment:
Same comments w.r.t. optional and signed integers

Reply:
How can we make the distinction without a comment here? Either optional or not, or signed or not, 0 still will have a special meaning in this system